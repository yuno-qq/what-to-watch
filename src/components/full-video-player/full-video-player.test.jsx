import React from 'react'
import renderer from 'react-test-renderer'
import {describe, test, expect} from "@jest/globals"

import FullVideoPlayer from "./full-video-player.jsx"


describe(`<FullVideoPlayer> snapshots:`, () => {
  test(`should render correctly`, () => {
    const movie = {
      imageSrc: `image.svg`,
      videoSrc: `video.wav`
    }

    const tree = renderer.create(<FullVideoPlayer
      duration={200}
      currentTime={20}
      movie={movie}
      renderItem={() => <div className="MOCK_COMPONENT"></div>}
    />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
