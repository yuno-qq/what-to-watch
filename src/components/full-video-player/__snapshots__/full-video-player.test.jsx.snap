// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullVideoPlayer> snapshots: should render correctly 1`] = `
<div
  className="player"
>
  <div
    className="MOCK_COMPONENT"
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          onClick={[Function]}
          value={10}
        />
        <button
          className="player__toggler"
          onMouseDown={[Function]}
          style={
            {
              "left": "10%",
            }
          }
        >
          Toggler
        </button>
      </div>
      <div
        className="player__time-value"
      >
        03:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      />
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
